update/update.go

```diff
@@ -30,38 +23,64 @@ func init() {
 // Cmd update
 var Cmd = &cobra.Command{
 	Use:     "update",
-	Short:   "Update sail to the latest release version : sail update",
+	Short:   "Update sail to the latest release version: sail update",
 	Long:    `sail update`,
 	Aliases: []string{"up"},
 	Run: func(cmd *cobra.Command, args []string) {
-		doUpdate(urlUpdateRelease, architecture)
+		doUpdate("", architecture)
 	},
 }
 
-var cmdUpdateSnapshot = &cobra.Command{
-	Use:     "snapshot",
-	Short:   "Update sail to latest snapshot version : sail update snapshot",
-	Long:    `sail update snapshot`,
-	Aliases: []string{"snap"},
-	Run: func(cmd *cobra.Command, args []string) {
-		doUpdate(urlUpdateSnapshot, architecture)
-	},
+func getURLArtifactFromGithub(architecture string) string {
+	client := github.NewClient(nil)
+	release, resp, err := client.Repositories.GetLatestRelease("runabove", "sail")
+	if err != nil {
+		fmt.Printf("Repositories.GetLatestRelease returned error: %v\n%v", err, resp.Body)
+		os.Exit(1)
+	}
+
+	if len(release.Assets) > 0 {
+		for _, asset := range release.Assets {
+			if *asset.Name == "sail-"+architecture {
+				return *asset.BrowserDownloadURL
+			}
+		}
+	}
+
+	fmt.Println("Invalid Artifacts on latest release. Please try in few minutes.")
+	fmt.Println("If it's persit, please open an issue on https://github.com/runabove/sail/issues")
+	os.Exit(1)
+	return ""
 }
 
 func doUpdate(baseurl, architecture string) {
-	url := fmt.Sprintf("%s/sail-%s", baseurl, architecture)
-
+	if architecture == "" {
+		fmt.Println("You seem to have a custom build of sail")
+		fmt.Println("Please download latest release on https://github.com/runabove/sail/releases")
+		os.Exit(1)
+	}
+	url := getURLArtifactFromGithub(architecture)
 	if internal.Verbose {
-		fmt.Fprintf(os.Stderr, "Debug: updating from %s\n", url)
+		fmt.Printf("Url to update sail: %s\n", url)
 	}
-
 	resp, err := http.Get(url)
```

Comment:
Can we add an expectation on the content type ? Or some heuristic testing like running "sail version" on the new version before replacing the current binary ? I've had fun side-effects with HTML binaries in the past :)